storage/local/interface.go

```diff
@@ -26,7 +26,8 @@ import (
 // Storage ingests and manages samples, along with various indexes. All methods
 // are goroutine-safe. Storage implements storage.SampleAppender.
 type Storage interface {
-	Querier
+	// Querier returns a new Querier on the storage.
+	Querier() (Querier, error)
```

Comment:
I don't think this has any requirement of something new being created. Case in point: `MemorySeriesStorage`, which just returns the wrapped regular storage.